### PR TITLE
doc/user: adjust Kafka `CONNECTION` options

### LINKED ISSUE #14979: docs: Update Kafka `CREATE SOURCE` documentation
### Feature request

The documentation for creating Kafka sources is a bit out dated and doesn't reflect the current syntax. https://materialize.com/docs/unstable/sql/create-source/kafka/
For example:
- `TOPIC` should be included with the other options.
- `WITH` has been replaced by `LEGACYWITH`

### PR DESCRIPTION
**Pull request by morsapaes**
Updating Kafka source options to match the new syntax. Fixes #14979.

Open questions:

* Do the defaults for `TOPIC METADATA REFRESH INTERVAL MS` and `FETCH MESSAGE MAX BYTES` still stand? I can't map the previously documented values to defaults in [`kafka_utils.rs`](https://github.com/MaterializeInc/materialize/blob/476f00cd4b389ecba523814531924a55c5672698/src/sql/src/kafka_util.rs#L81-L103);
* From #14656, it's unclear what other options we want to surface to users. @benesch @bkirwi, is there consensus on adding `PARTITION COUNT` and `RETENTION MS` (also `RETENTION BYTES`)? I've removed `STATISTICS INTERVAL MS` based on the discussion.

### CODE REVIEW

File: doc/user/content/sql/create-source/kafka.md [hunk 1]
```diff
@@ -359,7 +359,6 @@ CREATE VIEW jsonified_kafka_source AS
 ```sql
 CREATE SOURCE proto_source
   FROM KAFKA CONNECTION kafka_connection (TOPIC 'test_topic')
-  WITH (cache = true)
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/kafka.md:362]: Also couldn't figure out what happened to `cache`! Is it now parsed as `LEGACYWITH`?

**Review by bkirwi — COMMENTED** [review]
I think quite a few of these are still trapped in unsafe mode: https://github.com/MaterializeInc/materialize/blob/7e82ff67899f9887d3757c3c24c85a7b7aeff69a/src/sql/src/plan/statement/ddl.rs#L387-L393

I'm not sure if that means we should drop them from the docs for now? Anyways the docs themselves seem very correct!

**Comment by morsapaes** [comment]
Ah! I'd rather add options as we make them available, the question is whether the other options are supposed to be behind unsafe mode at this point. 🤔

**Comment by benesch** [comment]
I have a take on `TOPIC METADATA REFRESH INTERVAL MS`!

> `TOPIC METADATA REFRESH INTERVAL MS`: this one should also be an interval type, and should be permanently unsafe. It exists only so that our tests can notice new partitions faster than every 5m.

**Comment by morsapaes** [comment]
You're right — sorry I missed that, @benesch. Defaulting to only including the options marked as safe and tagging @nmeagan11 to double check that these are the ones we want to expose for now! 

Thank you both!

**Comment by nmeagan11** [comment]
I'm not sure we want to remove `client_id`? The rest look good!

**Comment by nmeagan11** [comment]
Summarizing a couple Slack conversations here - we decided to keep the options in unsafe for now and we will review one by one what we want to add back in as they come up with users. Said another way, this is all good for now!

**Pull request merged**